sdc/tests/test_dataframe.py

```diff
@@ -1369,10 +1370,12 @@ def test_impl(df, periods, method):
                            "E": [-1, np.nan, 1, np.inf],
                            "F": [np.nan, np.nan, np.inf, np.nan]})
         for periods in [0, 1, 2, 5, 10, -1, -2, -5]:
-            for method in [None, 'pad', 'ffill', 'backfill', 'bfill']:
-                result_ref = test_impl(df, periods, method)
-                result = hpat_func(df, periods, method)
-                pd.testing.assert_frame_equal(result, result_ref)
+            with self.subTest(periods=periods):
+                for method in [None, 'pad', 'ffill', 'backfill', 'bfill']:
+                    with self.subTest(method=method):
```

Comment:
Let's merge 2 usages of `subTest` to a single one:
`with self.subTest(periods=periods, method=method):`

and let's use `itertools.product`